debian/apport-general-hook.py

```diff
@@ -0,0 +1,62 @@
+import json
+import logging
+
+
+def get_ec2_data(instance_data):
+    """
+    provided cloud-init data from /run/cloud/instance-data.json
+    populate a smaller set of information helpful for debugging
+    """
+    stripped_data = {}
+    instance_identity_keys = (
+        "architecture",
+        "billingProducts",
+        "imageId",
+        "region",
+    )
+    for key in instance_identity_keys:
+        stripped_data[key] = instance_data["ds"]["dynamic"][
+            "instance-identity"
+        ]["document"][key]
+    stripped_data["instanceType"] = instance_data["ds"]["meta-data"][
+        "instance-type"
```

Comment:
This key is also found under ds.dynamic.instance-identity.instanceType if you wanted to include that in your for loop.
Examples of somewhat stale instance-data per cloud for reference: https://hackmd.io/aAJFerDRRNaoL0Q_vDRYBg#EC2

Reply:
Thanks. I refactored that code bit.